bin/validator_tests.py

```diff
@@ -0,0 +1,66 @@
+from collections.abc import Callable
+from typing import Optional, TypeVar
+
+_GENERATOR_NAMES: set[str] = set()
+GENERATORS: list[tuple[str, str | Callable[[str], Optional[str]], bool]] = []
```

Comment:
Maybe add documentation comments about what going on here? From what I'm gathering, these generators are created once `validator_tests` is imported, and these generators are then used to generate invalid test cases in `problem.py`.

One note about creating these generators in top-level code: this does mean that this is _always_ executed, also in the many instances where a command different than `bt validate` is called, and also when performing tab-completion in your shell. Since these generators are only used inside `Problem.validate_invalid_extra_data`, perhaps we should wrap this in a function anyway :stuck_out_tongue: I think Python will still us nest the decorator magic inside that function :slightly_smiling_face:

Reply:
> I think its just clearer this way and easier to extend.

It'll be just as easy to extend if it's not a global var, right? With the advantage that the creation of `GENERATORS` only happens when it's needed, and not every time any other `bt` command is used :slightly_smiling_face:

Reply:
well every other constant is also filled when bt is run even though they don't get used?